sourced/ml/extractors/roles_and_ids.py

```diff
@@ -0,0 +1,25 @@
+from typing import Iterable, Tuple
+
+import bblfsh
+
+from sourced.ml.algorithms.uast_to_role_id_pairs import Uast2RoleIdPairs
+from sourced.ml.extractors.bags_extractor import BagsExtractor
+from sourced.ml.algorithms import NoopTokenParser
+
+
+class RolesAndIdsExtractor(BagsExtractor):
+    """
+    Extractor wrapper for Uast2RoleIdPairs algorithm.
+    Note that this is unusual BagsExtractor since it returns iterable instead of bag.
+    """
+    NAMESPACE = ""
+    NAME = "roles and ids"
+    OPTS = {}
```

Comment:
You've just killed the "weight" option.

Reply:
An answer for three comments (this one and two next):
* This one
* https://github.com/src-d/ml/pull/205#discussion_r178544276
* https://github.com/src-d/ml/pull/205#discussion_r178544351

I did not use `@register_extractor` because it does not produce bags as others do. So nobody outside code will see it or use it directly. 
I make a note about it in comments:
> Note that this is unusual BagsExtractor since it returns iterable instead of bag.

So, I do not need any of `NAMESPACE `, "NAME" and "OPTS".

If you want I can split `BagsExtractor` into to abstract classes: `Extractor` and `BagsExtractor(Extractor)`. And my classes will be inherited from  `Extractor`.

I just thought it is a good idea at all, but do not have a lot of time for all this stuff.